### DISCUSSION TITLE
Sculpting the Future

### DISCUSSION THREAD

**Discussion by xynydev** [blue-build/.github#3] [General]
I'll try not to write too long of a post here, but here's my vision on the future plans for the project.

Short-term:
- [x] Bring in people
- [x] Make a small announcement

Mid-term:
- [x] Establish a website repo. This would serve as marketing, documentation, and tooling for custom image creation.
   - [x] Bring in ublue's make your own docs onto website.
   - [x] Create new content for website.
- [x] Bring in @gmpinder's rust cli project
- [x] Start work on [creating custom GitHub actions](https://docs.github.com/en/actions/creating-actions) to re-use in a template (that shouldn't be changing constantly)
- [ ] Do an official announcement tackling the "I thought this was supposed to be stable but now there's changes again" sentiment (make the change friendly, understandable, and clearly outline the small actions required by a developer)
- [ ] Prepare bling & startingpoint for the move, complete the move,
   - [x] Remove unnecessary features from bling (`/usr/bin/ublue-nix-install` etc.)
   - [x] Consolidate all bling content into `/modules`
- [x] Turn ublue making your own docs into a redirect for new project

Long-term:
- [x] Rework bling to function as a full-fledged module repo
- [x] Position the currently build.sh startingpoint as legacy, with the compiler-based templates as the strong recommendation
- [ ] Templates for other CI providers
- [ ] Alternative documentation for non-recipe based methods (writing Containerfiles & GitHub Actions yourself). Try not to compete with the possible (slightly planned) ublue docs outlining how their projects are built, how to fork them, etc.
- [ ] Support for other operating systems (at least VanillaOS and Distrobox os')
- [ ] Create an repo creator and configurator for the website (figuring out how to generate keys with WASM in browser would be cool)

**Comment by HexSleeves** [comment]
Would love to help anyway I can. Either through testing, or assisting in creating blue-build

**Comment by gmpinder** [comment]
If you know any Rust, take a look at the code, would be great to have
another person working on this with me.

https://gitlab.com/wunker-bunker/blue-build

Now mirrored here: https://github.com/blue-build/cli

**Reply by xynydev** [reply]
I can read rust and write a bit too, but checking Gitlab isn't unfortunately part of my routine (I check GitHub notifs atleast every few days) so it goes more unnoticed. I understand your preference for Gitlab, but it'd probably be beneficial to move it to where the rest of the project is, no?

I'm not the biggest fan of GitHub either lol, it's just convenience, legacy, and the network effect that drives me here.

**Reply by HexSleeves** [reply]
if you want, you can create a mirror to his repo on GitHub so it will fetch daily with updates. I will most likely setup a mirror just because I prefer github to gitlab

**Reply by xynydev** [reply]
> if you want, you can create a mirror to his repo on GitHub so it will fetch daily with updates.

Won't fetch MRs and Issues, though, right? That's the main way to keep track of projects and influence their development.

**Reply by tulilirockz** [reply]
Could also just move to either gitlab or github instead! Id prefer github because the main ublue org is on github, but idk!

**Reply by xynydev** [reply]
Moving to GH is the plan.

**Reply by gmpinder** [reply]
Repo is mirrored now https://github.com/blue-build/cli

**Comment by fiftydinar** [comment]
Long-term (or possibly shorter term) addition:

- Include `testing` branch in templates, which would allow users to test their changes before merging to `live` branch.
Make it in a way that it will be handled automatically, without potential conflicts.
I think that with BlueBuild tooling, this can be achieved cleanly.

**Reply by HexSleeves** [reply]
I think once we are feature parity with rpm-ostree/ starting point we should create a example repo under blue build that is a quick clone for users to use as well

**Reply by xynydev** [reply]
@n16hth4wkk that's what's on my agenda right after cleaning up bling

**Reply by gmpinder** [reply]
We could include the branch name in the tag if it's not running on the default branch. I might already be doing?

**Reply by fiftydinar** [reply]
> We could include the branch name in the tag if it's not running on the default branch. I might already be doing?

I think this is a good idea.

I wouldn't know the other usecase where users would want to change this manually.

When I had `testing` branch, I added sufix manually, so instead of `gidro-os`, it would be `gidro-os-testing`.

**Comment by gmpinder** [comment]
One of the things I've been working on is an `init`/`new` command that will create a new project for you to use with `bb`. I definitely think this will be useful for helping folks start out

**Reply by xynydev** [reply]
Sounds cool! Such a command should probably use a template repo as a base instead of bundling all the files in the program. I'll create one soon, once some other stuff is sorted out.
